src/getter.jl

```diff
@@ -0,0 +1,443 @@
+
+# Example
+For a four-momentum `(px, py, pz, E)`, this function returns `px^2 + py^2`.
+
+# Notes
+- The transverse components are the momentum projections in the x-y plane, perpendicular to the z-axis.
+
+# See Also
+- [`pt`](@ref): For the transverse momentum.
+"""
+pt2
+
+"""
+    pt(lv)
+
+Return the transverse momentum (``p_T``) of a given Lorentz-vector-like `lv`, defined as the Euclidean magnitude of the momentum components in the x-y plane.
+
+# Example
+For a four-momentum `(px, py, pz, E)`, this function returns `sqrt(px^2 + py^2)`.
+
+# Notes
+- The transverse components are defined with respect to the z-axis (3-axis), representing the perpendicular momentum to this axis.
+
+# See Also
+- [`pt2`](@ref): For the squared transverse momentum.
+
+"""
+pt
+
+"""
+    mt2(lv)
+
+Return the squared transverse mass (``m_T^2``) of a given Lorentz-vector-like `lv`, which is the difference between the squared time- (energy-) and the squared z-component.
+
+# Example
+For a four-momentum `(px, py, pz, E)`, this function returns `E^2 - pz^2`.
+
+# Notes
+- The transverse components are defined with respect to the z-axis (3-axis), indicating a projection in the x-y plane.
+
+# See Also
+- [`mt`](@ref): For the transverse mass.
+"""
+mt2
+
+"""
+    mt(lv)
+
+Return the transverse mass (``m_T``) of a given Lorentz-vector-like `lv`, calculated as the square root of the squared transverse mass. The transverse mass is often used in high-energy physics to describe the effective mass of a system when only the transverse components are considered.
+
+# Example
+For a four-momentum `(px, py, pz, E)`, this function returns `sqrt(E^2 - pz^2)`.
+
+# Notes
+- If the squared transverse mass (`mT²`) is negative, the function returns `-sqrt(-mT²)` to handle the imaginary mass situation that can occur in some relativistic systems.
+- The transverse components are defined with respect to the z-axis (3-axis), indicating momentum in the x-y plane.
+
+# Throws
+- May include a warning if the transverse mass is negative, depending on user settings.
+
+# See Also
+- [`mt2`](@ref): For the squared transverse mass.
+"""
+mt
+
+"""
+    eta(lv)
+
+Return the [pseudorapidity](https://en.wikipedia.org/wiki/Pseudorapidity) (``\\eta``) of a given Lorentz-vector-like `lv`. The pseudorapidity is a commonly used quantity in high-energy physics, particularly in collider experiments, and is defined as:
+
+```math
+    \\eta = -\\log(\\tan(\\theta/ 2))
+```
+
+where ``\\theta`` is the polar angle of the Lorentz-vector-like relative to the z-axis.
+
+
+# Example
+For a four-momentum `(px, py, pz, E)`, this function `log(tan(theta/2))` where `theta` is given by the [`polar_angle`](@ref).
+
+!!! warning
+
+    If the transverse momentum (`pt`) is zero (i.e., the particle is aligned with the beam axis), a warning is raised, and a large pseudorapidity value (±10e10) is returned as a convention. This occurs because the pseudorapidity is ill-defined when `pt = 0`.
```

Comment:
Is "10e10" intentionally chosen or should it be 1e11 or 1e10? Also it might be simpler/more straight forward to define these values as a general maximum rather than a special case? Otherwise if `pt` is very close to but not exactly 0 you might get a value larger than 1e11 and the user may be confused.
See also comments at the `eta` implementation.

Reply:
See my comment at the `eta` implementation ;-)